src/main/scala/pureconfig/ConfigConvert.scala

```diff
@@ -288,9 +288,13 @@ trait LowPriorityConfigConvertImplicits {
   }
 
   implicit val finiteDurationConfigConvert: ConfigConvert[FiniteDuration] = {
-    val fromString: String => FiniteDuration = (s: String) => DurationConvert.fromString(s, implicitly[ClassTag[FiniteDuration]]).get match {
-      case v if v.isFinite() => Duration(v.length, v.unit)
-      case _ => throw new IllegalArgumentException(s"Couldn't parse '$s' into a finite duration because it's infinite.")
+    val fromString: String => FiniteDuration = { (s: String) =>
+      DurationConvert.fromString(s, implicitly[ClassTag[FiniteDuration]])
+        .flatMap {
+          case d: FiniteDuration => Success(d)
+          case _ => Failure(new IllegalArgumentException(s"Couldn't parse '$s' into a FiniteDuration because it's infinite."))
+        }
+        .get
```

Comment:
I agree that it's weird. 
I don't relish the idea of making that breaking change to the API.
I could see an argument for adding `*Safe` versions of `fromString`, etc (e.g. `def fromStringSafe`) which took a `String => Try[T]`. On the other hand, that means a big expansion of the pureconfig API.
I'm pretty sure if you had both, you'd end up implementing the safe in in terms of the existing API because you have to protect against someone who thinks they're catching all the exceptions failing to catch all of them.
In short, I don't really care and can give an economist a run for his money on telling three sides of the story. 
@melrief, Do you have an opinion?